File: frontend/src/lib/roulette.ts
import { api } from "./api";
import type { MediaType } from "./media";

export type RouletteResult = {
  tmdb_id: number;
  media_type: MediaType;
  title: string;
  poster_url: string | null;
};

export async function spinRoulette(listId?: string): Promise<RouletteResult | null> {
  const { data } = await api.get<{ result: RouletteResult | null }>("/roulette", {
    params: listId ? { list_id: listId } : undefined,
  });
  return data.result;
}

File: backend/app/schemas/roulette.py
from typing import Literal

from pydantic import BaseModel


class RouletteResult(BaseModel):
    tmdb_id: int
    media_type: Literal["movie", "tv"]
    title: str
    poster_url: str | None


class RouletteResponse(BaseModel):
    result: RouletteResult | None
